roletracker/roletracker.py

```diff
@@ -0,0 +1,253 @@
+
+        self.config.register_role(**default_role)
+
+    async def initialize(self):
+        await self.register_casetypes()
+
+    @staticmethod
+    async def register_casetypes():
+        # register mod case
+        role_case = {
+            "name": "Role List Update",
+            "default_setting": True,
+            "image": "\N{PAGE FACING UP}",
+            "case_str": "roleupdate",
+        }
+        try:
+            await modlog.register_casetype(**role_case)
+        except RuntimeError:
+            pass
+
+    # Commands
+    @commands.group()
+    @commands.guild_only()
+    @checks.mod_or_permissions(manage_roles=True)
+    async def roletracker(self, ctx: GuildContext):
+        """Role Trackers settings"""
+        pass
+
+    @roletracker.command(name="set")
+    async def set_role(self, ctx: GuildContext, role: discord.Role, enabled: bool):
+        """
+        Sets role to be addable/removable
+        """
+        if not enabled:
+            if await self.config.role(role).USERS():
+                pred = MessagePredicate.yes_or_no(ctx)
+            await ctx.send(f"Found logs for role {role}, do you want to erase them?")
+            try:
+                await self.bot.wait_for("message", check=pred, timeout=30)
+            except asyncio.TimeoutError:
+                return await ctx.send("Timed out.")
+            if pred.result:
+                await self.config.role(role).USERS.set({})
+            await self.config.role(role).addable.set(False)
+            await ctx.tick()
+        else:
+            await self.config.role(role).addable.set(True)
+            await ctx.tick()
+
+    @roletracker.command(name="roles")
+    async def view_roles(self, ctx: GuildContext):
+        """
+        Lists all roles which can be added/removed
+        """
+        enabled = list()
+        for role in ctx.guild.roles:
+            if await self.config.role(role).addable() and not role.is_default():
+                enabled.append(role.name)
+        pages = pagify("\n".join(enabled))
+        await ctx.send("List of addable roles:")
+        for page in pages:
+            await ctx.maybe_send_embed(page)
+
+    @commands.bot_has_permissions(manage_roles=True)
+    @roletracker.command(name="add")
+    async def add_role(
+        self, ctx: GuildContext, member: discord.Member, role: discord.Role, *, reason: str = f"Added by {__cogname__}"
+    ):
+        """
+        Adds specified role to given user
+        """
+        if not await self.config.role(role).addable():
+            return await ctx.maybe_send_embed("Role isn't set as addable.")
+        try:
+            if role in member.roles:
+                return await ctx.maybe_send_embed("Member already has that role.")
+
+            data = await self.config.role(role).USERS()
+            if len(ctx.message.attachments):
+                attachment = ctx.message.attachments[0]
+                reason_message = f"{reason}. {attachment.url}"
```

Comment:
Reason message here and on line 121 should also add the role that was added too. Since they send in embeds, you can use `role.mention` so it looks nicer, it won't ping the role.

Reply:
Also include new lines between the reason, rule, and name, so something like:

```
reason_message = f"{reason}\nRole: {role.mention}\n{attachment.url}"
```